macros/models/_sector/transfers/erc20_stablecoins/transfers_erc20_stablecoins.sql

```diff
@@ -0,0 +1,150 @@
+            , CASE
+                WHEN t.blockchain = 'arbitrum' THEN ((effective_gas_price/1e9) * (gas_used/1e9))
+                WHEN t.blockchain = 'base' THEN ((gas_price/1e9) * (gas_used/1e9)) + (l1_fee/1e18)
+                WHEN t.blockchain = 'blast' THEN ((gas_price/1e9) * (gas_used/1e9)) + (l1_fee/1e18)
+                WHEN t.blockchain = 'optimism' THEN ((gas_price/1e9) * (gas_used/1e9)) + (l1_fee/1e18)
+                WHEN t.blockchain = 'scroll' THEN ((gas_price/1e9) * (gas_used/1e9)) + (l1_fee/1e18)
+                WHEN t.blockchain = 'tron' THEN ((gas_used) * (0.00042))
+                --WHEN t.blockchain = 'zkevm' THEN ((effective_gas_price/1e9) * (gas_used/1e9)) --zkevm uses effective_gas_price but this is not in table
+                ELSE ((tx.gas_price/1e9) * (tx.gas_used/1e9))
+            END AS tx_fee_gas_coin
+        FROM transfers t
+        INNER JOIN (
+            SELECT 
+                gas_price
+                , gas_used
+                , hash 
+                , CASE WHEN '{{blockchain}}' = 'arbitrum' THEN effective_gas_price ELSE NULL END AS effective_gas_price
+                , CASE
+                    WHEN '{{blockchain}}' = 'base' THEN l1_fee
+                    WHEN '{{blockchain}}' = 'blast' THEN l1_fee
+                    WHEN '{{blockchain}}' = 'optimism' THEN l1_fee
+                    WHEN '{{blockchain}}' = 'scroll' THEN l1_fee
+                    ELSE NULL
+                END AS l1_fee
+            FROM {{ source(blockchain, 'transactions') }}
+            {% if is_incremental() %}
+            WHERE block_time >= date_trunc('day', now() - interval '7' day)
+            {% endif %}
+        ) tx
+            ON tx.hash = t.evt_tx_hash 
+    )
+    , costs_and_transfers AS (
+        SELECT 
+            c.blockchain
+            , c.contract_address
+            , c.symbol
+            , c.evt_tx_hash
+            , c.evt_block_time
+            , c.evt_block_date
+            , c.tx_fee_gas_gwei
+            , c.stables_in_tx
+            , t.volume
+            , p.gas_coin
+            , p.median_price_day AS gas_coin_median_price_day
+        FROM tx_costs c
+        INNER JOIN (
+          SELECT
+                symbol
+                , evt_tx_hash
+                , volume
+            FROM transfer_volume
+        ) t
+            ON t.evt_tx_hash = c.evt_tx_hash 
+            AND t.symbol = c.symbol
+        LEFT JOIN ( -- Add on USD price of tx
+            SELECT 
+                CASE 
+                    WHEN symbol = 'ETH' THEN 'ethereum'
+                    WHEN symbol = 'MATIC' THEN 'polygon'
+                    WHEN symbol = 'XDAI' THEN 'gnosis'
+                    WHEN symbol = 'BNB' THEN 'bnb'
+                    WHEN symbol = 'AVAX' THEN 'avalanche_c'
+                    WHEN symbol = 'TRX' THEN 'tron'
+                    WHEN symbol = 'FTM' THEN 'fantom'
+                    WHEN symbol = 'CELO' THEN 'celo'
+                END AS chain
```

Comment:
if we're passing in a blockchain variable, we should be able to code this in a way to simplify this column logic rather than look at all chains?

Reply:
To my knowledge, the best price data for native tokens is where blockchain is `NULL` in the `prices.usd` table. So if I want to use that data I have to map them to a blockchain, which I thought was easiest by making the above case statement. 

What is the better solution to this problem?

Reply:
you are passing in `blockchain` value through the model to macro, so the transfers data above is already focused on one chain. we should keep the prices data focused to one chain as well to match, rather than all. you can filter on this variable already used, then also pass in another new variable for symbol as needed